File: apps/research-front/lib/src/domain/article.dart
import 'document.dart';
import 'fcc.dart';

class ArticleId {
  final String id;

  ArticleId(this.id);  
}

class Article {
  ArticleId id;
  ArticleOverview articleOverview;
  Document document;
  
    Article(this.id, this.articleOverview, this.document);
}

class ArticleOverview {
  String editor;
  String editorIcon;
  String title;
  String lastModified;
  String thumbnail;
  String description;

  ArticleOverview(this.editor, this.editorIcon, this.title, this.lastModified, this.thumbnail, this.description);
}

class Articles extends FCC<Article> {
  @override
  final List<Article> values;

  Articles(this.values);
  
}

File: apps/research-api/domain/article.go
package domain

import "time"

type ArticleId string

type Editor struct {
	Id   int
	Name string
	Icon string
}

type ThumbnailUrl string
type ThumbnailId string

type ArticleOverview struct {
	Title        string
	Editor       Editor
	LastModified time.Time
	Thumbnail    string
	Description  string
	//Tag Tag
}

type Article struct {
	ArticleId       ArticleId
	ArticleOverview ArticleOverview
	Content         map[string]interface{}
	Includes        map[string]interface{}
}

type Articles []Article
